src/acparser/__init__.py

```diff
@@ -44,54 +47,85 @@ def system(cmd):
 
 def main():
     "Entry point"
-    namespace = sys.argv[1]
-    collection_name = sys.argv[2]
-    collection_version = sys.argv[3]
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--tarfile", help="Path to the source tarball", required=True)
+    parser.add_argument(
+        "--namespace", help="Namespace of the collection", required=True
+    )
+    parser.add_argument("--name", help="Name of the collection", required=True)
+    parser.add_argument("--version", help="Version of the collection", required=True)
+    args = parser.parse_args()
+    namespace = args.namespace
+    collection_name = args.name
+    collection_version = args.version
+
+
+    EXISTS_GALAXY = False
+    REQUIRES_ANSIBLE = ""
+    LICENSE = ""
+    CHANGELOG = ""
+    REQUIREMENTS = []
+    COMMUNITY_COLLECTIONS = ""
+
+
     with tempfile.TemporaryDirectory() as collection_dir:
-        print("Downloading collection... \n")
-        system(
+        _, _, retcode = system(
             f"ansible-galaxy collection download -n -p {collection_dir} {namespace}.{collection_name}:{collection_version}"
         )
-        print("Collection downloaded. \n")
-        tarfilename = os.path.join(
-            collection_dir, f"{namespace}-{collection_name}-{collection_version}.tar.gz"
-        )
+        # check the return code
+        if retcode == 0:
+            tarfilename = os.path.join(
+                collection_dir,
+                f"{namespace}-{collection_name}-{collection_version}.tar.gz",
+            )
+            if os.path.exists(tarfilename):
+                EXISTS_GALAXY = True
+
+    # printing the output
+    if EXISTS_GALAXY:
+        print("Source exists in galaxy.")
+    else:
+        print("Source does not exist in galaxy.")
```

Comment:
Here's an example of what I was talking about when I said to change how `EXISTS_GALAXY` and other variables could be done:
```suggestion
        exists_galaxy = False
        if retcode == 0:
            tarfilename = os.path.join(
                collection_dir,
                f"{namespace}-{collection_name}-{collection_version}.tar.gz",
            )
            if os.path.exists(tarfilename):
                exists_galaxy = True

    # printing the output
    if exists_galaxy:
        print("Source exists in galaxy.")
    else:
        print("Source does not exist in galaxy.")
```